crates/arroyo-connectors/src/nats/source/mod.rs

```diff
@@ -422,7 +422,15 @@ impl NatsSourceFunc {
                                     }
                                 },
                                 Some(Err(msg)) => {
-                                    return Err(UserError::new("NATS message error", msg.to_string()));
+                                    match msg.kind() {
+                                        consumer::pull::MessagesErrorKind::MissingHeartbeat => {
```

Comment:
Is it worth failing after some number of missed heartbeats?

Reply:
That's a good idea, will have a think about how to best implement a check for that - probably something involving the NatsState cache